### DISCUSSION TITLE
How do I pass tenant level configuration items into a custom data source?

### DISCUSSION THREAD

**Discussion by catmanjan** [microsoft-search/pnp-modern-search#2609] [Q&A]
I need to be able to set a few configurable URLs once per tenant and have that setting flow through to a custom data source which extends BaseDataSource

What's the preferred way to get config items to the front end with pnp search?

**Comment by wobba** [comment]
You would need to find the API needed to read tenant config settings. I'm thinking this is best added as tokens to use and we can also expose them as context objects.

https://learn.microsoft.com/en-us/sharepoint/dev/spfx/tenant-properties?tabs=sprest
